nighly CI: registry test fails

https://github.com/linux-nvme/nvme-cli/actions/runs/28148159480/job/83359669028#step:4:1608

```
102/102 nvme-cli - registry-cli                                    FAIL             1.50s   exit status 1
>>> LD_LIBRARY_PATH=/nvme-cli/.build-ci/libnvme/src MALLOC_PERTURB_=194 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 MESON_TEST_ITERATION=1 /usr/bin/python3 /nvme-cli/.build-ci/../tests/nvme_registry_test.py ./nvme
```

```
================================== 102/102 ===================================
test:         nvme-cli - registry-cli
start time:   06:05:12
duration:     1.50s
result:       exit status 1
command:      LD_LIBRARY_PATH=/nvme-cli/.build-ci/libnvme/src MALLOC_PERTURB_=194 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 MESON_TEST_ITERATION=1 /usr/bin/python3 /nvme-cli/.build-ci/../tests/nvme_registry_test.py ./nvme
----------------------------------- stderr -----------------------------------
.....F...F...F
======================================================================
FAIL: test_retrieve_default_attr_is_owner (__main__.RegistryCLITest.test_retrieve_default_attr_is_owner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nvme-cli/.build-ci/../tests/nvme_registry_test.py", line 72, in test_retrieve_default_attr_is_owner
    result = self._run('registry', 'retrieve', '-d', 'nvme3')
  File "/nvme-cli/.build-ci/../tests/nvme_registry_test.py", line 51, in _run
    self.assertEqual(result.returncode, 0,
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
                     f'Command {cmd} failed:\n{result.stderr}')
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 1 != 0 : Command ['./nvme', 'registry', 'retrieve', '-d', 'nvme3'] failed:
--attr required


======================================================================
FAIL: test_retrieve_strips_dev_prefix (__main__.RegistryCLITest.test_retrieve_strips_dev_prefix)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nvme-cli/.build-ci/../tests/nvme_registry_test.py", line 77, in test_retrieve_strips_dev_prefix
    result = self._run('registry', 'retrieve', '-d', '/dev/nvme3')
  File "/nvme-cli/.build-ci/../tests/nvme_registry_test.py", line 51, in _run
    self.assertEqual(result.returncode, 0,
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
                     f'Command {cmd} failed:\n{result.stderr}')
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 1 != 0 : Command ['./nvme', 'registry', 'retrieve', '-d', '/dev/nvme3'] failed:
--attr required


======================================================================
FAIL: test_update_owner_is_rejected (__main__.RegistryCLITest.test_update_owner_is_rejected)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nvme-cli/.build-ci/../tests/nvme_registry_test.py", line 109, in test_update_owner_is_rejected
    self.assertNotEqual(result.returncode, 0)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 0 == 0
```

--- a/tests/nvme_registry_test.py
+++ b/tests/nvme_registry_test.py
@@ -43,7 +43,10 @@ class RegistryCLITest(unittest.TestCase):
 
     def _run(self, *args, expect_fail=False):
         cmd = [_NVME_BIN] + list(args)
+        # stdin is /dev/null so the owner-change confirmation prompt sees a
+        # non-interactive caller and proceeds without asking.
         result = subprocess.run(cmd, env=self.env,
+                                stdin=subprocess.DEVNULL,
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE,
                                 encoding='utf-8')
@@ -64,26 +67,27 @@ class RegistryCLITest(unittest.TestCase):
 
     def test_retrieve_returns_value(self):
         self._populate('nvme3', 'owner', 'stas')
-        result = self._run('registry', 'retrieve', '-d', 'nvme3', '-a', 'owner')
+        result = self._run('registry', 'retrieve', 'nvme3', '-a', 'owner')
         self.assertEqual(result.stdout.strip(), 'stas')
 
-    def test_retrieve_default_attr_is_owner(self):
+    def test_retrieve_requires_attr(self):
         self._populate('nvme3', 'owner', 'nbft')
-        result = self._run('registry', 'retrieve', '-d', 'nvme3')
-        self.assertEqual(result.stdout.strip(), 'nbft')
+        result = self._run('registry', 'retrieve', 'nvme3', expect_fail=True)
+        self.assertNotEqual(result.returncode, 0)
 
     def test_retrieve_strips_dev_prefix(self):
         self._populate('nvme3', 'owner', 'stas')
-        result = self._run('registry', 'retrieve', '-d', '/dev/nvme3')
+        result = self._run('registry', 'retrieve', '/dev/nvme3', '-a', 'owner')
         self.assertEqual(result.stdout.strip(), 'stas')
 
     def test_retrieve_missing_device_fails(self):
-        result = self._run('registry', 'retrieve', '-d', 'nvme99',
+        result = self._run('registry', 'retrieve', 'nvme99', '-a', 'owner',
                            expect_fail=True)
         self.assertNotEqual(result.returncode, 0)
 
     def test_retrieve_missing_device_arg_fails(self):
-        result = self._run('registry', 'retrieve', expect_fail=True)
+        result = self._run('registry', 'retrieve', '-a', 'owner',
+                           expect_fail=True)
         self.assertNotEqual(result.returncode, 0)
 
     # ------------------------------------------------------------------ #
@@ -91,26 +95,27 @@ class RegistryCLITest(unittest.TestCase):
     # ------------------------------------------------------------------ #
 
     def test_update_creates_entry(self):
-        self._run('registry', 'update', '-d', 'nvme5',
+        self._run('registry', 'update', 'nvme5',
                   '-a', 'note', '-V', 'boot-path SAN connection')
-        result = self._run('registry', 'retrieve', '-d', 'nvme5', '-a', 'note')
+        result = self._run('registry', 'retrieve', 'nvme5', '-a', 'note')
         self.assertEqual(result.stdout.strip(), 'boot-path SAN connection')
 
     def test_update_overwrites_entry(self):
         self._populate('nvme5', 'note', 'old note')
-        self._run('registry', 'update', '-d', 'nvme5',
+        self._run('registry', 'update', 'nvme5',
                   '-a', 'note', '-V', 'new note')
-        result = self._run('registry', 'retrieve', '-d', 'nvme5', '-a', 'note')
+        result = self._run('registry', 'retrieve', 'nvme5', '-a', 'note')
         self.assertEqual(result.stdout.strip(), 'new note')
 
-    def test_update_owner_is_rejected(self):
-        result = self._run('registry', 'update', '-d', 'nvme5',
-                           '-a', 'owner', '-V', 'stas', expect_fail=True)
-        self.assertNotEqual(result.returncode, 0)
-        self.assertIn('immutable', result.stderr)
+    def test_update_owner_succeeds_noninteractive(self):
+        # Writing the owner attribute is gated by a confirmation prompt, but a
+        # non-interactive caller proceeds without asking.
+        self._run('registry', 'update', 'nvme5', '-a', 'owner', '-V', 'stas')
+        result = self._run('registry', 'retrieve', 'nvme5', '-a', 'owner')
+        self.assertEqual(result.stdout.strip(), 'stas')
 
     def test_update_missing_args_fails(self):
-        result = self._run('registry', 'update', '-d', 'nvme5',
+        result = self._run('registry', 'update', 'nvme5',
                            expect_fail=True)
         self.assertNotEqual(result.returncode, 0)
 
@@ -120,13 +125,13 @@ class RegistryCLITest(unittest.TestCase):
 
     def test_delete_removes_entry(self):
         self._populate('nvme5', 'owner', 'stas')
-        self._run('registry', 'delete', '-d', 'nvme5')
-        result = self._run('registry', 'retrieve', '-d', 'nvme5',
+        self._run('registry', 'delete', 'nvme5')
+        result = self._run('registry', 'retrieve', 'nvme5', '-a', 'owner',
                            expect_fail=True)
         self.assertNotEqual(result.returncode, 0)
 
     def test_delete_nonexistent_fails(self):
-        result = self._run('registry', 'delete', '-d', 'nvme99',
+        result = self._run('registry', 'delete', 'nvme99',
                            expect_fail=True)
         self.assertNotEqual(result.returncode, 0)
 

